### ISSUE TITLE
Method parameter tooltip offers wrong class

### ISSUE THREAD

**Issue by robertyoudan** [eclipse-jdt/eclipse.jdt.ui#1471]
When mousing over a parameter type declaration, the tooltip offers up an unrelated class; not reachable by current package and imports and misses the correct class which is declared in the current file/class.
![eclipseerror](https://github.com/eclipse-jdt/eclipse.jdt.ui/assets/173324943/7944a421-7ba2-4195-bd67-fed9fd4f5d64)

**Comment by jjohnstn** [comment]
@robertyoudan Which version of Eclipse are you using?  I tried a similar generic abstract class with an inner static interface called Callback and I get the correct inner interface referred to.  When I do completion on Callback, I see the inner interface first and then the javax.security one after.   Are there any error markers in your file or are you using some add-on plug-ins?

**Comment by robertyoudan** [comment]
> @robertyoudan Which version of Eclipse are you using? I tried a similar generic abstract class with an inner static interface called Callback and I get the correct inner interface referred to. When I do completion on Callback, I see the inner interface first and then the javax.security one after. Are there any error markers in your file or are you using some add-on plug-ins?

Thanks for your reply. I believe it was the current version (4.32) but there's a small possibility it was 4.31. No build errors. I will try to reproduce today (using 4.32).

**Comment by robertyoudan** [comment]
Yep. Still does it with the current version. No build errors. Latest Temurin JDK (21.0.3.9-hotspot). Note when I get rid of the "..." it only finds the correct interface.

**Comment by iloveeclipse** [comment]
Ideally you should provide a simple project (with settings) demonstrating the problem.

**Comment by robertyoudan** [comment]
> Ideally you should provide a simple project (with settings) demonstrating the problem.

Believe me; if it was that simple I would have! This is part of a huge project so I will need to spend some time "de-coupling" the hierarchy,

**Comment by robertyoudan** [comment]
I just noticed another seemingly unrelated side effect. I have one codebase without the Callback parameter and one with it still included. The type hierarchy of an unrelated class has extra descendants that actually have NO RELATIONSHIP with the type hierarchy! See screenshot. Once the Callback class and parameters are removed, the erroneous descendants disappear!

![image](https://github.com/eclipse-jdt/eclipse.jdt.ui/assets/173324943/58c6c7af-815c-422a-8960-eb7dfd737212)
